File: binary-tree-level-order-traversal-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> emptyV;
        if(!root)return emptyV;
        vector<vector<int>> left = levelOrderBottom(root->left);
        vector<vector<int>> right = levelOrderBottom(root->right);
        int small = left.size()<right.size() ? left.size():right.size();
        int large = left.size()>right.size() ? left.size():right.size();
        vector<vector<int>> current(large+1);
        for(int i=0;i<left.size();i++)
        {
            int x = left.size() == small?large-small:0;;
            for(int j=0;j<left[i].size();j++)
                current[i+x].push_back(left[i][j]);
        }
        for(int i=0;i<right.size();i++)
        {
            int x = right.size() == small?large-small:0;;
            for(int j=0;j<right[i].size();j++)
                current[i+x].push_back(right[i][j]);
        }   
        
        current[large].push_back(root->val);
        return current;
    }
};

File: July LeetCoding Challenge/Binary Tree Level Order Traversal II.cs
public class Solution {
    private void Enqueue(Queue<TreeNode> q,TreeNode node)
    {
        if(node != null)
            q.Enqueue(node);
    }
    public IList<IList<int>> LevelOrderBottom(TreeNode root) {
        Queue<TreeNode> queue = new Queue<TreeNode>();
        List<IList<int>> res = new List<IList<int>>();
        Enqueue(queue,root);
        while(queue.Count > 0)
        {
            int size = queue.Count;
            List<int> row = new List<int>();
            for(int i=0;i<size;i++)
            {
                TreeNode node = queue.Dequeue();
                row.Add(node.val);
                Enqueue(queue,node.left);
                Enqueue(queue,node.right);
            }
            res.Add(row);
        }
        res.Reverse();
        return res;
    }
}
